1-development/post-rating/src/main/java/com/invoicexpress/postrating/service/PostService.java

```diff
@@ -0,0 +1,52 @@
+package com.invoicexpress.postrating.service;
+
+import com.invoicexpress.postrating.Enum.VoteEnum;
+import com.invoicexpress.postrating.entity.Post;
+import com.invoicexpress.postrating.exceptions.InvalidPostIdException;
+import com.invoicexpress.postrating.repository.PostRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+
+@Service
+public class PostService {
+
+    @Autowired
+    private PostRepository postRepository;
+
+    public Post updatePostVotes(int post_id, VoteEnum vote) throws InvalidPostIdException {
+
+        Post post = postRepository.findById(post_id).orElse(null);
+        if (post == null){
+            throw new InvalidPostIdException();
+
+        }
+        if (vote == VoteEnum.UP_VOTE) {
+            post.setUpVotes(post.getUpVotes() + 1);
+        } else {
+            post.setDownVotes(post.getDownVotes() + 1);
+        }
+        return postRepository.save(post);
+    }
+
+    public double calculateScoreOfPost(Post post){
+        float total = post.getDownVotes() + post.getUpVotes();
+        float upVotesOutTotal = post.getUpVotes()/total;
+        float confidentiality = 1.96f;
+
+        double score = (upVotesOutTotal + confidentiality*confidentiality/(2*total) - confidentiality * Math.sqrt(upVotesOutTotal * (1-upVotesOutTotal)
```

Comment:
In your opinion, this value should be final?

Reply:
Yes, this value could be final, since I will not change it.